pages/workoutlist.js

```diff
@@ -0,0 +1,176 @@
+import { useState } from "react";
+import styled from "styled-components";
+import { exercises } from "@/lib/exercises";
+import { workouts } from "@/lib/workouts";
+
+function findExerciseById(exerciseId) {
+  return exercises.find((exercise) => exercise.id === exerciseId);
+}
+
+export default function WorkoutsList() {
+  const [showDetails, setShowDetails] = useState({});
+
+  const toggleDetails = (workoutId) => {
+    setShowDetails((prev) => ({
+      ...prev,
+      [workoutId]: !prev[workoutId],
+    }));
+  };
+
+  return (
+    <>
+      <header>
+        <StickyHeader>Choose Your Workout</StickyHeader>
+      </header>
+
+      <Main>
```

Comment:
As mentioned in my previous comments, the `header` and `main` element should be part of your `Layout` component.